cognite/client/data_classes/simulators/simulators.py

```diff
@@ -0,0 +1,859 @@
+        return cls(
+            id=resource["id"],
+            external_id=resource["externalId"],
+            simulator_external_id=resource["simulatorExternalId"],
+            model_external_id=resource["modelExternalId"],
+            data_set_id=resource["dataSetId"],
+            file_id=resource["fileId"],
+            created_by_user_id=resource["createdByUserId"],
+            status=resource["status"],
+            created_time=resource["createdTime"],
+            last_updated_time=resource["lastUpdatedTime"],
+            version_number=resource["versionNumber"],
+            log_id=resource["logId"],
+            description=resource.get("description"),
+            status_message=resource.get("statusMessage"),
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return super().dump(camel_case=camel_case)
+
+
+class SimulatorRoutineRevision(CogniteResource):
+    def __init__(
+        self,
+        id: int,
+        external_id: str,
+        simulator_external_id: str,
+        routine_external_id: str,
+        simulator_integration_external_id: str,
+        model_external_id: str,
+        data_set_id: int,
+        created_by_user_id: str,
+        version_number: int,
+        created_time: int,
+        configuration: SimulatorRoutineConfiguration,
+        script: list[SimulatorRoutineStage],
+    ) -> None:
+        self.id = id
+        self.external_id = external_id
+        self.simulator_external_id = simulator_external_id
+        self.routine_external_id = routine_external_id
+        self.simulator_integration_external_id = simulator_integration_external_id
+        self.model_external_id = model_external_id
+        self.data_set_id = data_set_id
+        self.created_by_user_id = created_by_user_id
+        self.version_number = version_number
+        self.created_time = created_time
+        self.configuration = configuration
+        self.script = script
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            id=resource["id"],
+            external_id=resource["externalId"],
+            simulator_external_id=resource["simulatorExternalId"],
+            routine_external_id=resource["routineExternalId"],
+            simulator_integration_external_id=resource["simulatorIntegrationExternalId"],
+            model_external_id=resource["modelExternalId"],
+            data_set_id=resource["dataSetId"],
+            created_by_user_id=resource["createdByUserId"],
+            version_number=resource["versionNumber"],
+            created_time=resource["createdTime"],
+            configuration=SimulatorRoutineConfiguration._load(resource["configuration"], cognite_client),
+            script=[SimulatorRoutineStage._load(stage_, cognite_client) for stage_ in resource["script"]],
+        )
```

Comment:
example from timeseries, looks much simpler:

![Screenshot 2024-11-06 at 16 37 44](https://github.com/user-attachments/assets/b279dab1-e7fb-4203-8e38-24d336d454e9)

Reply:
script and configutation I agree, but simple props maybe can be loaded automatically?